iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/WeakHashAlgorithmsCheck.java

```diff
@@ -0,0 +1,151 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.annotation.CheckForNull;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Argument;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S4790")
+public class WeakHashAlgorithmsCheck implements IacCheck {
+
+  private static final String MESSAGE = "Using weak hashing algorithms is security-sensitive.";
+  private static final Set<String> OPENSSL_SENSITIVE_SUBCOMMAND = Set.of("md5", "sha1", "rmd160", "ripemd160");
+  private static final Set<String> OPENSSL_SENSITIVE_DGST_OPTION  = Set.of("-md2", "-md4", "-md5", "-sha1", "-ripemd160", "-ripemd", "-rmd160");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, WeakHashAlgorithmsCheck::checkRun);
+  }
+
+  private static void checkRun(CheckContext ctx, RunInstruction runInstruction) {
+    CommandDetector sensitiveOpenSslSubcommand = CommandDetector.builder()
+      .with("openssl"::equals)
+      .with(OPENSSL_SENSITIVE_SUBCOMMAND::contains)
+      .build();
+    CommandDetector sensitiveOpenSslDgst = CommandDetector.builder()
+      .with("openssl"::equals)
+      .with("dgst"::equals)
+      .withOptional(s -> !OPENSSL_SENSITIVE_DGST_OPTION.contains(s) && s.startsWith("-"))
+      .with(OPENSSL_SENSITIVE_DGST_OPTION::contains)
+      .build();
+
+    sensitiveOpenSslSubcommand.search(runInstruction.arguments())
+      .forEach(sensitiveOpensslCall -> ctx.reportIssue(TextRanges.mergeRanges(sensitiveOpensslCall), MESSAGE));
+    sensitiveOpenSslDgst.search(runInstruction.arguments())
+      .forEach(sensitiveOpensslCall -> ctx.reportIssue(TextRanges.mergeRanges(sensitiveOpensslCall), MESSAGE));
+  }
+
+  private static class CommandDetector {
+
+    List<Predicate<String>> predicates;
+    List<Boolean> isOptionalPredicates;
+
+    private CommandDetector(List<Predicate<String>> predicates, List<Boolean> isOptionalPredicates) {
+      this.predicates = predicates;
+      this.isOptionalPredicates = isOptionalPredicates;
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    /**
+     * Return all block of arguments which match the list of predicates.
+     */
+    private List<List<Argument>> search(List<Argument> arguments) {
+      List<ArgumentResolution> resolvedArgument = arguments.stream().map(ArgumentResolution::of).collect(Collectors.toList());
```

Comment:
I can imagine merging [SONARIAC-647](https://sonarsource.atlassian.net/browse/SONARIAC-647) could bring some value here. When the resolution contains the argument we don't have to resolve this every time.

Reply:
Yes :D I would like to move forward on this, it would really simplify things.
I did understand you already have the code done for this ? Otherwise I can also create the PR, let me know.